src/mds/Server.cc

```diff
@@ -3299,6 +3299,30 @@ CDentry* Server::prepare_stray_dentry(MDRequestRef& mdr, CInode *in)
   return straydn;
 }
 
+CDir* Server::get_quota_root(CDir *dir) {
+    auto cur = dir;
+    while (true) {
+        if (cur->inode->is_root()) {
+            return cur;
+        }
+        if (cur->inode->get_projected_inode()->quota.is_enable()) {
+            return cur;
+        }
+        cur = cur->get_parent_dir();
```

Comment:
@vshankar Additionally, I confirmed the cur can be null in this line, so I am implementing to fix this issue.